modules/generic_worker/manifests/init.pp

```diff
@@ -0,0 +1,105 @@
+        user_homedir     => $user_homedir,
+        bugzilla_api_key => $bugzilla_api_key,
+    }
+
+    $livelog_certificate = "${user_homedir}/livelog.crt"
+    $livelog_key         = "${user_homedir}/livelog.key"
+    $task_dir            = "${user_homedir}/tasks"
+    $caches_dir          = "${user_homedir}/caches"
+    $downloads_dir       = "${user_homedir}/downloads"
+    $opengpg_signing_key = "${user_homedir}/generic-worker.openpgp.signing.key"
+    $ed25519_signing_key = "${user_homedir}/generic-worker.ed25519.signing.key"
+
+    exec {
+        'create opengpg signing key':
+            path    => ['/bin', '/sbin', '/usr/local/bin', '/usr/bin'],
+            user    => $user,
+            cwd     => $user_homedir,
+            command => "generic-worker new-openpgp-keypair --file ${opengpg_signing_key}",
+            unless  => "test -f ${opengpg_signing_key}",
+            require => Class['packages::generic_worker'];
+
+        'create ed25519 signing key':
+            path    => ['/bin', '/sbin', '/usr/local/bin', '/usr/bin'],
+            user    => $user,
+            cwd     => $user_homedir,
+            command => "generic-worker new-ed25519-keypair --file ${ed25519_signing_key}",
+            unless  => "test -f ${ed25519_signing_key}",
+            require => Class['packages::generic_worker'];
+    }
+
+    case $::operatingsystem {
+        'Darwin': {
+
+            $reboot_command = '/usr/bin/sudo /sbin/reboot'
+
+            file {
+                '/Library/LaunchAgents/net.generic.worker.plist':
+                    ensure  => present,
+                    content => template('generic_worker/generic-worker.plist.erb'),
+                    mode    => '0644',
+                    owner   => $::root_user,
+                    group   => $::root_group;
+
+                '/usr/local/bin/run-generic-worker.sh':
+                    ensure  => present,
+                    content => template('generic_worker/run-generic-worker.sh.erb'),
+                    mode    => '0755',
+                    owner   => $::root_user,
+                    group   => $::root_group;
+
+                '/etc/generic-worker.config':
+                    ensure  => present,
+                    content => template('generic_worker/generic-worker.config.erb'),
+                    mode    => '0644',
+                    owner   => $::root_user,
+                    group   => $::root_group;
+            }
+
+            service { 'net.generic.worker':
+                require => File['/Library/LaunchAgents/net.generic.worker.plist'],
+                enable  => true,
+            }
+
+            host { $taskcluster_host:
+                ip => '127.0.0.1'
+            }
+
+            httpd::config { 'proxy.conf':
+                content => template('generic_worker/proxy-httpd.conf.erb'),
+            }
+        }
+        default: {
+            fail("${module_name} is not supported on ${::operatingsystem}")
+        }
+    }
+}
```

Comment:
This class was modified. See https://github.com/mozilla-releng/build-puppet/pull/409/files